cmd/ndm_daemonset/controller/blockdevicestore_test.go

```diff
@@ -168,9 +168,11 @@ func TestUpdateDevice(t *testing.T) {
 
 	// Retrieve disk resource
 	cdevR, err := fakeController.GetBlockDevice(fakeDeviceUID)
-
-	devR.ObjectMeta.Name = "disk-updated-fake-uuid"
-	err = fakeController.UpdateBlockDevice(devR, cdevR)
+	if err != nil {
+		t.Fatal(err)
+	}
+	cdevR.ObjectMeta.Name = "disk-updated-fake-uuid"
+	err = fakeController.UpdateBlockDevice(*cdevR, nil)
```

Comment:
Why is this change required?

Reply:
From the error log -
`If the resource is not present then it should return error.`
My understanding - there maybe 2 cases -
1. If the resource is not present we can get error while fetching it.
2. If the resource is not present we can get if we try to update it.

Earlier it was returning error for case 2 as was getting name and namespace from new object. Now we change that we are taking name and namespace from the older object so it will not fail.
Now it can fail for case 1 only and for that, we need to pass the old object as nil.